runtime/parachains/src/runtime_api_impl/v1.rs

```diff
@@ -233,25 +233,20 @@ pub fn session_index_for_child<T: initializer::Config>() -> SessionIndex {
 }
 
 /// Implementation for the `AuthorityDiscoveryApi::authorities()` function of the runtime API.
+/// It is a heavy call, but currently only used for authority discovery, so it is fine.
 /// Gets current and historical authority ids using session_info module.
-pub fn get_current_and_historical_authority_ids<T: initializer::Config>() -> Vec<AuthorityDiscoveryId> {
+pub fn get_historical_authority_ids<T: initializer::Config>() -> Vec<AuthorityDiscoveryId> {
```

Comment:
The API of this function has now changed and it can include duplicates. 

This function should include the call to `AuthorityDiscovery::next_authorities` which can be achieved by adding an extra bound on `T` if necessary (although I believe it is implied by `initializer::Config`.

Reply:
@rphmeier I am slightly confused right now. Is `session_info[current_session_index].discovery_keys` is equivalent to `AutorityDiscovery::next_authorities()` in every case? In that case, I can directly do `earliest_session..=current_session`, instead of calling  `AuthorityDiscovery::next_authorities`.

Reply:
No, that's not what I said at all. Please re-review my comment! `session_info[current_session_index].discovery_keys` is the same as `AuthorityDiscovery::current_authorities`. Why would it be otherwise? This is the only thing that makes sense.

Reply:
@rphmeier Ok. Makes sense. I have updated code to reflect this.